core/lib/rom/schema.rb

```diff
@@ -138,7 +138,13 @@ def self.define(name, attributes: EMPTY_ARRAY, attr_class: Attribute, **options)
 
     # @api private
     def self.attributes(attributes, attr_class)
-      attributes.map { |type| attr_class.new(type) }
+      attributes.map do |attr|
+        if attr.is_a?(Hash)
+          attr_class.new(attr[:type], **attr[:options] || {})
```

Comment:
I believe this can be simplified like this:

``` ruby
 attr_class.new(attr[:type], attr.fetch(:options, EMPTY_HASH))
```

WDYT?

Reply:
Sure, much nicer. Done!